### PR TITLE
BoringSSL compatilibity fixes

### PR DESCRIPTION
**Pull request by nachooya**

### CODE REVIEW

**Comment by lgrahl** [comment]
Thanks. Could you elaborate on the use case of using BoringSSL instead of OpenSSL? I'm asking since the BoringSSL project itself states:

> Although BoringSSL is an open source project, it is not intended for general use, as OpenSSL is. We don't recommend that third parties depend upon it. Doing so is likely to be frustrating because there are no guarantees of API or ABI stability.

**Comment by nachooya** [comment]
We use BoringSSL for our project because its integrates easily on Android NDK. Works without any problem. Thanks.

File: src/librawrtc/diffie_hellman_parameters.c [hunk 1]
```diff
@@ -3,6 +3,11 @@
 #include <openssl/err.h> // ERR_clear_error
 #include <openssl/pem.h> // PEM_read_bio_DHparams
 #include <openssl/ssl.h> // SSL_CTX_set_tmp_dh, SSL_CTX_set_ecdh_auto
+
+#ifdef OPENSSL_IS_BORINGSSL
+#include <limits.h>
```
Comment (lgrahl) [inline src/librawrtc/diffie_hellman_parameters.c]: I guess this is an oversight not limited to BoringSSL.

**Comment by lgrahl** [comment]
Moved to #139 due to the restructuring of the source files.

**Pull request closed without merging**